test/local/routes/sms.js

```diff
@@ -359,3 +362,203 @@ describe('/sms disabled', () => {
   })
 })
 
+describe('/sms/status', () => {
+  let log, config, geodb, geodbResult, routes, route, request
+
+  beforeEach(() => {
+    log = mocks.spyLog()
+    config = {
+      sms: {
+        enabled: true,
+        regions: /^US$/
+      }
+    }
+    geodb = sinon.spy(() => geodbResult)
+    routes = makeRoutes({ log, config }, { '../geodb': () => geodb })
+    route = getRoute(routes, '/sms/status')
+    request = mocks.mockRequest({
+      credentials: {
+        email: 'foo@example.org'
+      },
+      log: log
+    })
+  })
+
+  describe('sms.status returns isOk:true and getGeoData returns US', () => {
+    let response
+
+    beforeEach(() => {
+      sms.status = sinon.spy(() => P.resolve({ isOk: true }))
+      geodbResult = Promise.resolve({ location: { countryCode: 'US' } })
+      return runTest(route, request)
+        .then(r => response = r)
+    })
+
+    it('returned the correct response', () => {
+      assert.deepEqual(response, { ok: true })
+    })
+
+    it('called log.begin correctly', () => {
+      assert.equal(log.begin.callCount, 1)
+      const args = log.begin.args[0]
+      assert.equal(args.length, 2)
+      assert.equal(args[0], 'sms.status')
+      assert.equal(args[1], request)
+    })
+
+    it('called sms.status correctly', () => {
+      assert.equal(sms.status.callCount, 1)
+      assert.equal(sms.status.args[0].length, 0)
+    })
+
+    it('called geodb correctly', () => {
+      assert.equal(geodb.callCount, 1)
+      const args = geodb.args[0]
+      assert.equal(args.length, 1)
+      assert.equal(args[0], request.app.clientAddress)
+    })
+
+    it('did not call log.error', () => {
+      assert.equal(log.error.callCount, 0)
+    })
+  })
+
+  describe('sms.status returns isOk:false and getGeoData returns US', () => {
+    let response
+
+    beforeEach(() => {
+      sms.status = sinon.spy(() => P.resolve({ isOk: false }))
+      geodbResult = Promise.resolve({ location: { countryCode: 'US' } })
+      return runTest(route, request)
+        .then(r => response = r)
+    })
+
+    it('returned the correct response', () => {
+      assert.deepEqual(response, { ok: false })
+    })
+
+    it('called log.begin once', () => {
+      assert.equal(log.begin.callCount, 1)
+    })
+
+    it('called sms.status once', () => {
+      assert.equal(sms.status.callCount, 1)
+    })
+
+    it('did not call geodb', () => {
+      assert.equal(geodb.callCount, 0)
+    })
+
+    it('did not call log.error', () => {
+      assert.equal(log.error.callCount, 0)
+    })
+  })
+
+  describe('sms.status returns isOk:true and getGeoData returns CA', () => {
```

Comment:
> Think it makes sense to abstract the different country tests?

Although I'm in favour of abstracting setup and teardown, I see abstracted assertions as a massive anti-pattern. They make it harder for the casual reader to quickly grok the invariants and they encourage exhaustive testing, which can obscure where the important boundary conditions are covered.

I much prefer to only assert the important stuff, and to do it all longhand so that readers can quickly understand what the edge cases are.

^imho